internal/controller/stages/stages.go

```diff
@@ -43,6 +44,10 @@ func ReconcilerConfigFromEnv() ReconcilerConfig {
 	return cfg
 }
 
+type appHealthEvaluator interface {
+	EvaluateHealth(context.Context, kargoapi.FreightReference, []kargoapi.ArgoCDAppUpdate) *kargoapi.Health
+}
```

Comment:
Would go away if we did [this](https://github.com/akuity/kargo/pull/1807/files#r1558305789).